src/aio/aio_theme_switch.py

```diff
@@ -61,118 +59,60 @@ def __init__(
         - ThemeSwitchAIO.ids.store(aio_id)
         """
 
-        from dash_bootstrap_templates import load_figure_template
-
-        load_figure_template(dbc_themes_lowercase)
-
-        if aio_id is None:
-            aio_id = str(uuid.uuid4())
-
-        if themes is None:
-            themes = [dbc.themes.BOOTSTRAP, dbc.themes.CYBORG]
+        # init icons and switch_props
         if icons is None:
             icons = {"left": "fa fa-moon", "right": "fa fa-sun"}
-
-        switch_props = switch_props.copy()
-        if "value" not in switch_props:
-            switch_props["value"] = True
-        if "className" not in switch_props:
-            switch_props["className"] = "d-inline-block ms-1"
+        if switch_props is None:
+            switch_props = {}
+        # set "value" and "className" if they don't exist
+        switch_props.setdefault("value", True)
+        switch_props.setdefault("className", "d-inline-block ms-1")
+
+        # add fontawesome resource for the icons and dbc.css to apply Bootstrap theme to dcc, DataTable and Dash AG Grid
+        app = get_app()
+        app.config.external_stylesheets += [
+            "https://use.fontawesome.com/releases/v5.15.4/css/all.css",
+            # "https://cdn.jsdelivr.net/gh/AnnMarieW/dash-bootstrap-templates/dbc.min.css"
+        ]
+
+        # if using custom themes in /assets, filter them out to not be automatically imported by Dash
+        # and let the switch handle them.
+        # append "|" if assets_ignore has already regex rules
+        if app.config.assets_ignore:
+            app.config.assets_ignore += "|"
+        # add the themes to ignore in /assets
+        if isinstance(themes, str):
+            app.config.assets_ignore += f'{themes.replace("/assets/", "")}'
+        else:
+            app.config.assets_ignore += f'{themes[0].replace("/assets/", "")}|{themes[1].replace("/assets/", "")}'
```

Comment:
With the way the clientside callback works, I don't think this is necessary

Reply:
Indeed the clientside callback should clean the existing themes, but I think it's cleaner to not let Dash add the themes in /assets which will be removed afterwards.
I refactored a little this part, but I can remove it completely if you prefer 👍

Reply:
That's a good point, no harm in keeping it.
What if someone configs the assets file to be something other than the default "assets"?

Reply:
haha you're right! I'll take a look at that 👍

Reply:
Actually it was a though one! as we can have a custom folder on server side with `assets_folder` and on client side with `assets_url_path`
That took more time than expected but I think I found something working 💪